src/Extractor/HttpClient.php

```diff
@@ -263,7 +263,12 @@ public function fetch($url, $skipTypeVerification = false, $httpHeader = [])
         }
 
         // remove utm parameters & fragment
-        $effectiveUrl = preg_replace('/((\?)?(&(amp;)?)?utm_(.*?)\=[^&]+)|(#(.*?)\=[^&]+)/', '', rawurldecode($effectiveUrl));
```

Comment:
Interestingly, the regex also handles `&amp;`-separated queries. Not sure why it did that. One possible sources (`getMetaRefreshURL`) decodes them:

https://github.com/j0k3r/graby/blob/e1154098851dd347002314445c18847655f5a289/src/Extractor/HttpClient.php#L529

Reply:
Maybe `getUglyURL`?

Reply:
I don't think so, `getUglyURL` and `getMetaRefreshURL` calls, should they return anything, would lead to the exit of the method before reaching my change:

https://github.com/j0k3r/graby/blob/e1154098851dd347002314445c18847655f5a289/src/Extractor/HttpClient.php#L257-L263

However @j0k3r is there any reason to have handled `&amp;` here?

Reply:
I think it was just in case we have that kind of given url

Reply:
So should we continue to handle them?

Reply:
I think so, is it hard to achieve?

Reply:
FullTextRSS seems to have `preg_replace('/(\?|\&)utm_[a-z]+=[^\&]+/', '', $url);`

That line in graby was added during the first commit (👴), I've no idea why I've added the `&amp;`.

Maybe you just avoid a regression, you can do:

```php
$uri = new Uri(str_replace('&amp;', '&', $effectiveUrl));
```

Reply:
I've included your suggested change @j0k3r